control/VersionedRequestFilter.php

```diff
@@ -32,7 +32,7 @@ public function preRequest(SS_HTTPRequest $request, Session $session, DataModel
 			$dummyController->popCurrent();
 			// Prevent output in testing
 			if(class_exists('SapphireTest', false) && SapphireTest::is_running_test()) {
-				return false;
```

Comment:
I think in 4.x we could probably push up the "testing case exemption" up to director. If we allow request filters to throw SS_HTTPResponses normally, it wouldn't be a problem, but that would be an API breakage in 3.x.